gordo/machine/model/anomaly/diff.py

```diff
@@ -159,14 +208,16 @@ def cross_validate(
         # Depend on having the trained fold models
         kwargs.update(dict(return_estimator=True, cv=cv))
 
-        cv_output = cross_validate(self, X=X, y=y, **kwargs)
+        cv_output = c_val(self, X=X, y=y, **kwargs)
 
         self.feature_thresholds_per_fold_ = pd.DataFrame()
         self.aggregate_thresholds_per_fold_ = {}
         self.smooth_feature_thresholds_per_fold_ = pd.DataFrame()
         self.smooth_aggregate_thresholds_per_fold_ = {}
         smooth_aggregate_threshold_fold = None
         smooth_tag_thresholds_fold = None
+        if self.window is not None and self.smoothing_method is None:
```

Comment:
I think these two lines should be moved into `__init__` method, because in this way `self.smoothing_method` stay `None` until `cross_validation()` method be called

Reply:
Thanks, good idea. Moved the whole `if` statement it to `__init__()`, agree this is the best solution. Also had to adjust tests accordingly.